src/core/components/resolve.js

```diff
@@ -1,64 +1,82 @@
 'use strict'
 
-const promisify = require('promisify-es6')
 const isIpfs = require('is-ipfs')
-const setImmediate = require('async/setImmediate')
 const CID = require('cids')
+const nodeify = require('promise-nodeify')
 const { cidToString } = require('../../utils/cid')
 
-module.exports = (self) => {
-  return promisify(async (name, opts, cb) => {
-    if (typeof opts === 'function') {
-      cb = opts
-      opts = {}
-    }
+/**
+ * @typedef { import("../index") } IPFS
+ */
 
-    opts = opts || {}
+/**
+ * @typedef {Object} ResolveOptions
+ * @prop {string} cidBase - Multibase codec name the CID in the resolved path will be encoded with
+ * @prop {boolean} [recursive=true] - Resolve until the result is an IPFS name
```

Comment:
Is the recursive parameter implemented? It seems from the code like it's assumed to always be true

Reply:
right now this is just to be passed to ipns